generator/src/main/java/com/scottlogic/deg/generator/generation/combinationstrategies/PinningCombinationStrategy.java

```diff
@@ -29,25 +30,26 @@
 public class PinningCombinationStrategy implements CombinationStrategy {
 
     @Override
-    public Stream<DataBag> permute(Stream<Stream<DataBag>> dataBagSequences) {
+    public Stream<DataBag> permute(Stream<Supplier<Stream<DataBag>>> dataBagSequences) {
         Iterable<DataBag> iterable = new PinningCombinationStrategy
                 .InternalIterable(dataBagSequences);
 
         return StreamSupport.stream(iterable.spliterator(), false);
     }
 
     class InternalIterable implements Iterable<DataBag> {
-        private final Stream<Stream<DataBag>> dataBagSequences;
+        private final Stream<Supplier<Stream<DataBag>>> dataBagSequences;
 
-        InternalIterable(Stream<Stream<DataBag>> dataBagSequences) {
+        InternalIterable(Stream<Supplier<Stream<DataBag>>> dataBagSequences) {
             this.dataBagSequences = dataBagSequences;
         }
 
         @Override
         public Iterator<DataBag> iterator() {
-            List<SequenceAndBaselineTuple> tuples = this.dataBagSequences
-                    .map(sequence -> new SequenceAndBaselineTuple(sequence.iterator()))
-                    .collect(Collectors.toList());
+            List<SequenceAndBaselineTuple> tuples = dataBagSequences
+                .map(Supplier::get)
+                .map(sequence -> new SequenceAndBaselineTuple(sequence.iterator()))
```

Comment:
Not needing a change, but I would do this as three maps:
`.map(Supplier::get).map(BaseStream::iterator).map(SequenceAndBaselineTuple::new)`